models/share.php

```diff
@@ -11,6 +11,12 @@ public function add(){
 		$post = filter_input_array(INPUT_POST, FILTER_SANITIZE_STRING);
 
 		if($post['submit']){
```

Comment:
As this if is only a "security" check if it the form was submitted, please do not nest ifs if not needed. In this case the check for submit can be done on start where you check if there is no submit key then return/skip this method:
```suggestion
		if(!$post['submit']) {
		    return;
		}
```